lib/client.js

```diff
@@ -18,70 +18,110 @@ var GiantSwarm = (function () {
 
   return {
 
-    _authorizedRequest: function() {
-      if (authToken) {
-        return request.set('Authorization', 'giantswarm ' + authToken);
+    // to be thrown in case of bad parameters/arguments/values given
+    UserException: function(message) {
+      this.message = message;
+      this.name = "UserException";
+      this.toString = function(){
+        return this.name + ": " + this.message;
+      };
+    },
+
+    // simplify validation of string parameters for internal methods
+    _validateStringParameters: function(parameters) {
+      for (name in parameters) {
+        if (!parameters[name]) {
+          throw new this.UserException("Parameter '"+ name +"' must be given");
+        }
+        if (typeof(parameters[name]) !== 'string') {
+          throw new this.UserException("Parameter '"+ name +"' must be of type string");
+        }
```

Comment:
I am not sure about throwing exceptions. I think proper error handling would be good to eventually handle actions to cleanup things or notify users with animated clicki bunti. So it would be cool to return errors and let the caller decide what to do with that. I am curious to hear other opinions on that.